src/components/accent-color.js

```diff
@@ -0,0 +1,36 @@
+/* global AFRAME */
+
+AFRAME.registerComponent('accent-color', {
+  schema: {
+    color: {
+      type: 'color'
+    },
+    index: {
+      type: 'int'
+    }
+  },
+  init: function () {
+    this.meshes = [];
+    // The group children aren't ready when the component is initialized, so wait a bit.
+    // TODO: Find a better way to time this.
+    setTimeout(() => {
```

Comment:
This is a workaround since the entity children aren't visible when the component is initialized. I'm curious which event is fired that can be used to time this correctly.

Reply:
`gltf-model` component fires a `model-loaded` event, so you probably want to listen to that.

```js
if (this.el.getObject3D('mesh')) {
  // do your thing
} else {
  this.el.addEventListener('model-loaded', () => {
    this.meshes.length = 0; // be sure to not keep previous gtlf meshes if we switched the model
    // do your thing
  })`
}
```

I wouldn't keep meshes in an array on the component, that's source of issues, for example if you remove the gtlf-model component, you currently keep meshes in the array that aren't garbage collected. I would just do the traverse in update, without storing an array of meshes on the component.

Reply:
Sounds good, I tried the `model-loaded` and it works perfect!

Reply:
The `this.el.getObject3D('mesh')` check is needed if the accent-color component is added later, so after you got that model-loaded event. gltf-model set the loaded glb with `el.setObject3D('mesh', model)`